terminus/geometry/circle.py

```diff
@@ -45,45 +45,66 @@ def __eq__(self, other):
     def __hash__(self):
         return hash((self.center, self.radius))
 
+    def radius(self):
+        return self.radius
+
+    def center(self):
+        return self.center
+
     def almost_equal_to(self, other, decimals=5):
         return self.center.almost_equal_to(other.center, decimals) and abs(self.radius - other.radius) < 1e-5
 
-    def intersection(self, other):
-        d = (other.center - self.center).norm()
-        r1 = self.radius
-        r2 = other.radius
-        if d == 0 and r1 == r2:
+    def intersection(self, other, buffer=1e-7, decimals=7):
+        p = self.center
+        q = other.center
+        A = self.radius
+        B = other.radius
+        """
+        To find the intersection, we solve the two equations system given by:
+        (Point(x, y) - p).norm_squared = A ** 2
+        (Point(x, y) - q).norm_squared = B ** 2
+        """
+        M = 2 * (q.x - p.x)
+        N = 2 * (q.y - p.y)
+        R = A ** 2 - B ** 2 + q.norm_squared() - p.norm_squared()
+        # M * x + N * y = R
+        if self.almost_equal_to(other, decimals):
             return [self]
-        if d == 0 and r1 != r2:
+        if p == q and A != B:
             return []
-        if (d > self.radius + other.radius) or (d < abs(self.radius - other.radius)):
+        if abs(N) < buffer:
+            x = R / M
+            if abs(x - p.x) < A and abs(x - q.x) < B:
```

Comment:
For those that have actual meaning, it would be nice to generate temp variables. That would make the code clearer and reduce the amount of times you repeat a computation. As an example `x - p.x` or `A ** 2` is done many places in the code.